src/components/Heading/Heading.tsx

```diff
@@ -47,7 +46,7 @@ export class Heading extends React.PureComponent<Props, State> {
 
   constructor(props: Props) {
     super(props);
-    this.state = {tagName: props.appearance.toLowerCase()};
+    this.state = {tagName: (props.appearance.toLowerCase()) as TagName};
```

Comment:
@shlomitc isn't that a hack? the `.toLowerCase()` breaks the type back to string otherwise

Reply:
Not sure what's the best pattern here.
maybe worth asking a typescript guru.

Reply:
will write that for me to ask someone tomorrow, but since we are doing it in other places already I will merge this one for now. If there will be a better suggestion that I will get tomorrow, I will change it